### PR TITLE
GODRIVER-1844 export unified test runner

### PR DESCRIPTION
**Pull request by iwysiu**
All of the functions that were not calling the runner were made private. Notes:
1) matches_test.go got broken up into its functions and its test
2) Some of the variables in EntityMap were renamed to avoid collisions between the variables and the now unexported functions.

### CODE REVIEW

**Comment by divjotarora** [comment]
@iwysiu There's a number of test failures in the linked patch build. Can you look into those before we review?

**Comment by iwysiu** [comment]
> @iwysiu There's a number of test failures in the linked patch build. Can you look into those before we review?

Should be fixed! As a note, I suspect that there will be further changes for the atlas planned maintenance stuff, but I figured this was a big enough change to get looked at on its own.

**Review by benjirewis — APPROVED** [review]
LGTM! Just to clarify, we do want to export `integration/unified/unified_spec_test` and not `integration/unified_spec_test`, right?

**Review by divjotarora — COMMENTED** [review]
Can you explain how the planned maintenance test runner provides test files to run? Does it give you one filepath at a time, a path to a directory of files, or something else?

**Comment by iwysiu** [comment]
> Can you explain how the planned maintenance test runner provides test files to run? Does it give you one filepath at a time, a path to a directory of files, or something else?

The test runner says "json blob" but it looks to me like the ruby implementation reads a file

File: mongo/integration/unified/unified_spec_runner.go [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+	Description       string                      `bson:"description"`
+	SchemaVersion     string                      `bson:"schemaVersion"`
+	RunOnRequirements []mtest.RunOnBlock          `bson:"runOnRequirements"`
+	CreateEntities    []map[string]*entityOptions `bson:"createEntities"`
+	InitialData       []*collectionData           `bson:"initialData"`
+	TestCases         []*testCase                 `bson:"tests"`
+}
+
+// RunTestDirectory runs the files in the given directory, which must be in the unifed spec format
+func RunTestDirectory(t *testing.T, directoryPath string) {
```
Comment (divjotarora) [inline mongo/integration/unified/unified_spec_runner.go:81]: Why does this need to be exported?

Comment (iwysiu) [reply mongo/integration/unified/unified_spec_runner.go]: done

**Review by divjotarora — APPROVED** [review]

**Pull request merged**